diff --git a/src/basic/vx_log/string_array.cc b/src/basic/vx_log/string_array.cc
index ee823b8e0..190faea42 100644
--- a/src/basic/vx_log/string_array.cc
+++ b/src/basic/vx_log/string_array.cc
@@ -812,7 +812,7 @@ regex_t buffer;
 regex_t *preg = &buffer;
 
 // Check for null pointers
-if( !reg_exp_str || !test_str ) return false;
+if( !reg_exp_str || !test_str || strlen(reg_exp_str) == 0 || strlen(test_str) == 0) return false;
 
 if( regcomp(preg, reg_exp_str, REG_EXTENDED*REG_NOSUB) != 0 ) {
    mlog << Error << "\ncheck_reg_exp(char *, char *) -> "
